src/scr.c

```diff
@@ -1708,22 +1713,39 @@ int SCR_Route_file(const char* file, char* newfile)
 
     /* build absolute path to file */
     scr_path* path_abs = scr_path_from_str(file);
-    if (! scr_path_is_absolute(path_abs)) {
-      /* the path is not absolute, so prepend the current working directory */
-      char cwd[SCR_MAX_FILENAME];
-      if (scr_getcwd(cwd, sizeof(cwd)) == SCR_SUCCESS) {
-        scr_path_prepend_str(path_abs, cwd);
-      } else {
-        /* problem acquiring current working directory */
-        scr_abort(-1, "Failed to build absolute path to %s @ %s:%d",
-          file, __FILE__, __LINE__
-        );
+    if (scr_preserve_directories) {
+      if (! scr_path_is_absolute(path_abs)) {
+        /* the path is not absolute, so prepend the current working directory */
+        char cwd[SCR_MAX_FILENAME];
+        if (scr_getcwd(cwd, sizeof(cwd)) == SCR_SUCCESS) {
```

Comment:
Doesn't prefix also default to cwd? I think if someone sets SCR_PREFIX, they may be surprised to find their files somewhere else.